### PR TITLE
Fetch selfhosted profile

### PR DESCRIPTION
**Pull request by hypest**
Adds the XMLRPC call and new event to fetch the selfhosted site's user's profile.

Some of the profile data are persisted locally, namely the `email` and `display name`. These are the data currently needed to populate the WPAndroid's login epilogue screen so, only those are persisted.

The profile fetch is triggered by the `FETCH_PROFILE_XML_RPC` Action. Data are fetched from the site and saved locally as part of the `SiteModel`. Upon local DB update, a `OnSiteChanged` event is emitted.

TODO: Update https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/Actions-and-OnChanged-events when this PR gets merged.

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/site/SiteXMLRPCClient.java [hunk 1]
```diff
@@ -39,6 +39,43 @@ public SiteXMLRPCClient(Dispatcher dispatcher, RequestQueue requestQueue, UserAg
...
+                        site.error = error;
+                        mDispatcher.dispatch(SiteActionBuilder.newFetchedProfileXmlRpcAction(site));
+                    }
+                }
+        );
+
+        add(request);
+    }
+
+    private SiteModel profileResponseToAccountModel(Object response, SiteModel site) {
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/site/SiteXMLRPCClient.java]: Could you group this method with the other private ones, underneath the `public` network call methods? Just to keep things consistent as far as ordering with other client classes.

**Comment by aforcier** [comment]
How about adding a connected test to [ReleaseStack_SiteTestXMLRPC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/01c735b50372a0debacf8e5075768dec9b5ca576/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestXMLRPC.java)?

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/site/SiteXMLRPCClient.java (hunk 1 above)
Comment (hypest) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/site/SiteXMLRPCClient.java]: Sure! Done in 1d6e962.

**Comment by hypest** [comment]
Thanks for picking this up for review @aforcier !

I added a connected test with d460a47. Let me know if something more is needed on that front.

Ready for another pass!

**Comment by aforcier** [comment]
Looks good @hypest - just a merge conflict resolution needed and this is good to go 👍

**Comment by hypest** [comment]
Conflict with `develop` fixed. Ready for another pass!

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**